src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java

```diff
@@ -182,6 +185,24 @@ public String getInlineN1qlQuery() {
 		return StringUtils.hasText(query) ? query : null;
 	}
 
+	/**
+	 * is this a 'delete'?
+	 *
+	 * @return is this a 'delete'?
+	 */
+	public boolean isDeleteQuery() {
+		return getName().toLowerCase().startsWith("delete");
```

Comment:
should we use Locale.ROOT in the toLowerCase?

Reply:
done on exists, delete and count.  Mark suggested an annotation.  I will (eventually) see what mongo does.